### PR TITLE
CI: use native oldest for package

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Let's do not make PT assumptions/constraints, it shall automatically use PT 1.9 for now, as it is min requirement for the package

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda

### CODE REVIEW

**Review by otaj — APPROVED** [review]
Don't we need to also do something on lines 104-105?

**Review by akihironitta — COMMENTED** [review]
I believe some of the job names also change, so we need to update the checkgroup.yml.

File: .github/workflows/ci-pytorch-test-full.yml [hunk 1]
```diff
@@ -95,14 +95,17 @@ jobs:
     - name: Pull legacy checkpoints
       run: bash .actions/pull_legacy_checkpoints.sh
 
+    - name: Adjust PyTorch versions in requirements files
+      if: matrix.requires != 'oldest'
```
[the remark this answers is not in the document]
Comment (carmocca) [reply .github/workflows/ci-pytorch-test-full.yml:99]: It will do that even in the oldest job, because pytorch-version will be 1.9

**Review by awaelchli — APPROVED** [review]

File: .github/workflows/ci-pytorch-test-full.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Borda) [reply .github/workflows/ci-pytorch-test-full.yml:99]: if we will keep then align, but if miss it we would have silently installed different version

[the remark this answers is not in the document]
Comment (carmocca) [reply .github/workflows/ci-pytorch-test-full.yml:99]: Such misalignment should be covered by 

```python
        python -c "import torch; assert torch.__version__.startswith('${{ matrix.pytorch-version }}')"
```

in line 146

[the remark this answers is not in the document]
Comment (Borda) [reply .github/workflows/ci-pytorch-test-full.yml:99]: yes, unless you push it to be aligned by this code block:
```
python ./requirements/pytorch/adjust-versions.py requirements/pytorch/base.txt ${{ matrix.pytorch-version }}
```
so in other words it negates the min replacement, right?

[the remark this answers is not in the document]
Comment (carmocca) [reply .github/workflows/ci-pytorch-test-full.yml:99]: It does not negate anything. If you run locally

```py
python .actions/assistant.py replace_oldest_ver
python ./requirements/pytorch/adjust-versions.py requirements/pytorch/base.txt 1.9
git diff requirements/pytorch/base.txt
```

You'll see it correctly sets

```python
torch==1.9.1
```

[the remark this answers is not in the document]
Comment (Borda) [reply .github/workflows/ci-pytorch-test-full.yml:99]: Yes, if this CI PT version is identical to min version in requirements, but the situation I want to prevent is that we update requirements and miss this CI that the CI would be running the wrong version; see for example:
```
python .actions/assistant.py replace_oldest_ver
python ./requirements/pytorch/adjust-versions.py requirements/pytorch/base.txt 1.10
git diff requirements/pytorch/base.txt
```
But in principle, the situation is requirements are updated to >=1.11, but CI still say 1.9, then we would run CI with 1.9 instead of 1.11

[the remark this answers is not in the document]
Comment (Borda) [reply .github/workflows/ci-pytorch-test-full.yml:99]: this si really just prevention issues cause by human in the loop :otter:

[the remark this answers is not in the document]
Comment (carmocca) [reply .github/workflows/ci-pytorch-test-full.yml:99]: Okay. Resolving
